src/browser/modules/D3Visualization/lib/visualization/components/Visualization.ts

```diff
@@ -17,216 +17,113 @@
-
-import * as vizRenderers from '../renders/init'
+import { BaseType, Selection, select as d3Select } from 'd3-selection'
+import { D3ZoomEvent, ZoomBehavior, zoom as d3Zoom } from 'd3-zoom'
+
+import { ZOOM_SCALE_EXTENT } from '../constants'
+import {
+  node as nodeRenderer,
+  relationship as relationshipRenderer
+} from '../renders/init'
 import { nodeMenuRenderer } from '../renders/menu'
+import { ForceSimulation } from './ForceSimulation'
 import Graph from './Graph'
 import GraphGeometry from './GraphGeometry'
 import Relationship from './Relationship'
 import VizNode from './VizNode'
-import { Layout } from './layout'
+import {
+  nodeEventHandlers,
+  relationshipEventHandlers,
+  zoomEventHandler
+} from './mouseEventHandlers'
 import GraphStyle from 'browser/modules/D3Visualization/graphStyle'
 
 export type MeasureSizeFn = () => { width: number; height: number }
-export type VizObj = {
-  style: any
-  trigger: (_event: any, ..._args: any[]) => void
-  zoomInClick: (el: any) => { zoomInLimit: boolean; zoomOutLimit: boolean }
-  zoomOutClick: (el: any) => { zoomInLimit: boolean; zoomOutLimit: boolean }
-  boundingBox: () => void
-  resize: () => void
-  update: (options: {
-    updateNodes: boolean
-    updateRelationships: boolean
-    precompute?: boolean
-  }) => void
-}
 
-const noOp = () => undefined
-const vizFn = function (
-  el: SVGElement,
-  measureSize: MeasureSizeFn,
-  graph: Graph,
-  layout: Layout,
-  style: GraphStyle
-): VizObj {
-  const viz: VizObj = {
-    style,
-    // to be overwritten externally
-    trigger: noOp,
-    // to be defined later in this file, adding now for type safety
-    zoomInClick: () => ({
-      zoomInLimit: false,
-      zoomOutLimit: false
-    }),
-    zoomOutClick: () => ({
-      zoomInLimit: false,
-      zoomOutLimit: false
-    }),
-    boundingBox: noOp,
-    resize: noOp,
-    update: noOp
-  }
+export class Visualization<T extends Element = SVGElement> {
+  private root: Selection<T, unknown, BaseType, unknown>
+  private baseGroup: Selection<SVGGElement, unknown, BaseType, unknown>
+  private rect: Selection<SVGRectElement, unknown, BaseType, unknown>
+  private container: Selection<SVGGElement, unknown, BaseType, unknown>
+  private geometry: GraphGeometry
+  private zoomBehavior: ZoomBehavior<SVGGElement, unknown>
 
-  const root = d3Select(el)
-  const baseGroup = root.append('g').attr('transform', 'translate(0,0)')
-  const rect = baseGroup
-    .append('rect')
-    .style('fill', 'none')
-    .style('pointer-events', 'all')
-    // Make the rect cover the whole surface, center of the svg viewbox is in (0,0)
-    .attr('x', () => -Math.floor(measureSize().width / 2))
-    .attr('y', () => -Math.floor(measureSize().height / 2))
-    .attr('width', '100%')
-    .attr('height', '100%')
-    .attr('transform', 'scale(1)')
-
-  const container = baseGroup.append('g')
-  const geometry = new GraphGeometry(style)
+  public forceSim: ForceSimulation
```

Comment:
As far as I understand it class properties are public by default, so I'm not sure if it adds anything to explicitly annotate when we don't do it in the rest of the code base. But it's a matter of taste, so feel free to keep it

Reply:
Personally I prefer the more explicit version, but I didn't think about the rest of the codebase, will update the code to be more consistent with everything else.